src/duplicate.coffee

```diff
@@ -1,7 +1,181 @@
 # Description:
-#   A hubot script to do nothing
+#   A hubot script to duplicate messages between services
+#
+# Configuration:
+#   HUBOT_DUPLICATE_TO
+#   HUBOT_DUPLICATE_FROM
+#   HUBOT_DUPLICATE_ROOMS
+#
+# Commands:
+#   hubot duplicate about - Echoes an expanded about message
 #
 # Author:
 #   Andrew Lucas (sqweelygig) <andrewl@resin.io> <sqweelygig@gmail.com>
+class Duplicator
+	constructor: (robot) ->
+		@to =
+			name: process.env.HUBOT_DUPLICATE_TO
+			adapter: require('hubot-' + process.env.HUBOT_DUPLICATE_TO).use(robot)
+		@from = { name: process.env.HUBOT_DUPLICATE_FROM }
+		@pairs = JSON.parse process.env.HUBOT_DUPLICATE_ROOMS
```

Comment:
This seems to be used later only for lookup: why don't we require it to be an object? I.e. atm it looks like it has to be something like:

```javascript
pairs = [ { 'flowdock': 'roomB', 'gitter': 'roomA' } ]
```

It could be:

```javascript
roomMapping = { 'roomB': 'roomA' }
```

I don't think we need the flowdock/gitter signifiers, afaict, because there's only one to/from.

The code in `@echo` loops over `@pairs` but only uses the last option it finds, so it's not like we're trying to support duplicate (heh) mappings. With this, the `@echo` code goes from:

```
for pair in @pairs when pair[@from.name] is from_ids.flow
    to_ids.flow = pair[@to.name]
```

to

```
to_ids.flow = @roomMapping[from_ids.flow]
```

which is way clearer. I'm lacking information because there's no documentation on the env vars - is there a reason this won't work?